drivers/meter/ade7754/ade7754.c

```diff
@@ -0,0 +1,1156 @@
+		break;
+	case ADE7754_CURRENT_PHASE_B:
+		ret = ade7754_update_bits(dev, ADE7754_REG_WAVMODE, ADE7754_WAVSEL_MSK,
+					  no_os_field_prep(ADE7754_WAVSEL_MSK, ADE7754_CURRENT_PHASE_B));
+		break;
+	case ADE7754_CURRENT_PHASE_A:
+		ret = ade7754_update_bits(dev, ADE7754_REG_WAVMODE, ADE7754_WAVSEL_MSK,
+					  no_os_field_prep(ADE7754_WAVSEL_MSK, ADE7754_CURRENT_PHASE_A));
+		break;
+	case ADE7754_VOLTAGE_PHASE_C:
+		ret = ade7754_update_bits(dev, ADE7754_REG_WAVMODE, ADE7754_WAVSEL_MSK,
+					  no_os_field_prep(ADE7754_WAVSEL_MSK, ADE7754_VOLTAGE_PHASE_C));
+		break;
+	case ADE7754_VOLTAGE_PHASE_B:
+		ret = ade7754_update_bits(dev, ADE7754_REG_WAVMODE, ADE7754_WAVSEL_MSK,
+					  no_os_field_prep(ADE7754_WAVSEL_MSK, ADE7754_VOLTAGE_PHASE_B));
+		break;
+	default:
+		ret = ade7754_update_bits(dev, ADE7754_REG_WAVMODE, ADE7754_WAVSEL_MSK,
+					  no_os_field_prep(ADE7754_WAVSEL_MSK, ADE7754_VOLTAGE_PHASE_A));
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Select the gain of the current channels inputs.
+ * @param dev - The device structure.
+ * @param sel - gain value
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_current_gain(struct ade7754_dev *dev,
+			 enum ade7754_pga_gain sel)
+{
+	int ret;
+
+	switch (sel) {
+	case ADE7754_PGA_GAIN_4:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA1_MSK,
+					  no_os_field_prep(ADE7754_PGA1_MSK, ADE7754_PGA_GAIN_4));
+		break;
+	case ADE7754_PGA_GAIN_2:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA1_MSK,
+					  no_os_field_prep(ADE7754_PGA1_MSK, ADE7754_PGA_GAIN_2));
+		break;
+	default:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA1_MSK,
+					  no_os_field_prep(ADE7754_PGA1_MSK, ADE7754_PGA_GAIN_1));
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Select the gain of the voltage channels inputs.
+ * @param dev - The device structure.
+ * @param sel - gain value
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_voltage_gain(struct ade7754_dev *dev,
+			 enum ade7754_pga_gain sel)
+{
+	int ret;
+
+	switch (sel) {
+	case ADE7754_PGA_GAIN_4:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA2_MSK,
+					  no_os_field_prep(ADE7754_PGA2_MSK, ADE7754_PGA_GAIN_4));
+		break;
+	case ADE7754_PGA_GAIN_2:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA2_MSK,
+					  no_os_field_prep(ADE7754_PGA2_MSK, ADE7754_PGA_GAIN_2));
+		break;
+	default:
+		ret = ade7754_update_bits(dev, ADE7754_REG_GAIN, ADE7754_PGA2_MSK,
+					  no_os_field_prep(ADE7754_PGA2_MSK, ADE7754_PGA_GAIN_1));
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Select the formula used for active energy calculation
+ * @param dev - The device structure.
+ * @param sel - formula value
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade7754_active_energy_formula(struct ade7754_dev *dev,
+				  enum ade7754_watm sel)
+{
+	int ret;
+
+	switch (sel) {
```

Comment:
you can drop the switch and do this directly:
```
{
    return ade7754_update_bits(dev, ADE7754_REG_WATMODE, ADE7754_WATM_MSK,
					  no_os_field_prep(ADE7754_WATM_MSK, sel));
}
```

I would also add some input error handling on that sel to make sure it's within bounds before the function call. But if by applying a mask on it it limits it to the correct range it's supposed to be, you can skip that completely.

Reply:
applies to many other places throughout the file